src/nvim/ex_getln.c

```diff
@@ -2263,20 +2280,56 @@ static void draw_cmdline(int start, int len)
     msg_outtrans_len(ccline.cmdbuff + start, len);
 }
 
+void ui_ext_cmdline_char(int c, int shift)
```

Comment:
Consistency is good. But "consistency" is not an argument for doing X consistently over doing Y consistently. And we _consistently_ try to not extract short functions that only will be called once and are heavily coupled to their surroundings.

And marshallers are not equivalent to extracting functions. Extracting functions manually adds yet another layer, located in a different place, one must analyse where values can be passed trough or module-globals maybe are  used. Marshallers are sematically equal to calling `ui_event` directly.